projects/fsastorefrontlib/src/cms-components/my-account/inbox/inbox-tab/inbox-messages/inbox-messages.component.ts

```diff
@@ -116,7 +116,10 @@ export class InboxMessagesComponent implements OnInit, OnDestroy {
       if (!msg.read && msg.uid === message.uid) {
         msg.read = true;
         this.subscription.add(
-          this.inboxService.setMessagesState([msg.uid], true).subscribe()
+          this.inboxService
+            .setMessagesState([msg.uid], true)
+            .pipe(map(() => this.inboxService.setUnreadMessageState(msg.read)))
```

Comment:
I think `tap` would be more appropriate here since we are not changing the stream